pkg/util/push/otlp.go

```diff
@@ -90,6 +98,69 @@ func OTLPHandler(overrides *validation.Overrides, cfg distributor.OTLPConfig, so
 	})
 }
 
+func decodeOTLPWriteRequest(r *http.Request, maxRecvMsgSize int) (pmetricotlp.ExportRequest, error) {
```

Comment:
Most of this comes from https://github.com/prometheus/prometheus/blob/789c9b1a5e455850ed9b3c89cafb37df75ce1e50/storage/remote/codec.go#L873
Which you are not using. because of 2 lines.
 
Maybe the proper solution is to implement this limit in prometheus too :) as a follow up. I leave it up to you.

Reply:
Thanks for letting me know.
But I'm not sure the Prometheus needs this logic. The Prometheus remote write handlers do not have the limiting logic.

Reply:
@friedrichg 
I re-implemented `decodeOTLPWriteRequest` func using the existing `util.ParseProtoReader`.